tests/test_tasks.py

```diff
@@ -77,3 +78,118 @@ def test_active_learning_discrete(model, dataset):
          'num_initial_points': 10}
 
     run_AL_disc(ConvertToNamespace(d), is_test=True)
+
+
+# Testing modified regression and classification runs with mocks
+
+class RegressionMock(object):
+    """
+    Regression mock.
+    """
+    def fit(self, X: np.ndarray, Y:np.ndarray) -> None:
+        pass
+    def predict(self, X: np.ndarray) -> (np.ndarray, np.ndarray):
+        mu = np.array([[1., 2., 3.], [4., 5., 6.]])
+        var = np.array([[.1, .2, .3], [.4, .5, .6]])
+        return mu, var
+
+class ApproximateRegressionMock(RegressionMock):
+    """
+    Approximate regression mock.
+    """
+    def predict(self, X: np.ndarray) -> (np.ndarray, np.ndarray):
+        mu = np.array([[[1., 2., 3.], [4., 5., 6.]], [[1.5, 2.5, 3.5], [4.5, 5.5, 6.5]]])
```

Comment:
can you explain in the code why these classes return these specific shapes, please.